lib/waterline/model/lib/associationMethods/update.js

```diff
@@ -65,7 +65,11 @@ var Update = module.exports = function(collection, proto, mutatedModels, cb) {
     var reduced = nestedOperations.reduceAssociations(collection.identity, collection.waterline.schema, vals);
     _values = _.merge(_values, reduced);
   });
-  
+
+  if (!mutatedModels.length) {
+    return cb(null, proto);
```

Comment:
This is just to ignite a discussion. This is wrong to begin with, because simple updates don't have mutatedModels.